apps/frontend/components/auth/SignUp/SignUpVerifyEmail.tsx

```diff
@@ -0,0 +1,226 @@
+import { useSignUpModalStore } from '@/stores/signUpModal'
+import { valibotResolver } from '@hookform/resolvers/valibot'
+import { useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import * as v from 'valibot'
+import { AuthMessage } from '../AuthMessage'
+import { SignUpApi } from './api'
+
+interface VerifyEmailInput {
+  email: string
+  verificationCode: string
+}
+
+const schema = v.object({
+  verificationCode: v.pipe(
+    v.string(),
+    v.length(6, 'Code must be 6 characters long')
+  )
+})
+
+const TIME_LIMIT = 300
+
+export function SignUpVerifyEmail() {
+  const [endTime, setEndTime] = useState(() => Date.now() + TIME_LIMIT * 1000)
+  const [remaining, setRemaining] = useState(() =>
+    Math.max(0, Math.round((endTime - Date.now()) / 1000))
+  )
+
+  const [codeExpired, setCodeExpired] = useState(false)
+  const [emailVerified, setEmailVerified] = useState(false)
+  const { nextModal, setFormData, formData } = useSignUpModalStore(
+    (state) => state
+  )
+  const [codeError, setCodeError] = useState<string | null>(null)
+  const [emailAuthToken, setEmailAuthToken] = useState('')
+
+  const {
+    handleSubmit,
+    register,
+    getValues,
+    trigger,
+    formState: { errors }
+  } = useForm<VerifyEmailInput>({
+    resolver: valibotResolver(schema)
+  })
+
+  useEffect(() => {
+    if (codeExpired || emailVerified) {
+      return
+    }
+    const id = setInterval(() => {
+      const rem = Math.max(0, Math.round((endTime - Date.now()) / 1000))
+      setRemaining(rem)
+      if (rem === 0) {
+        setCodeExpired(true)
+      }
+    }, 1000)
+    return () => clearInterval(id)
+  }, [codeExpired, emailVerified])
+
+  const renderTimer = () => {
+    const minutes = Math.floor(remaining / 60)
+    const seconds = remaining % 60
+    return `${minutes}:${seconds < 10 ? '0' : ''}${seconds}`
+  }
+
+  const onSubmit = (data: VerifyEmailInput) => {
+    setFormData({
+      ...data,
+      headers: {
+        'email-auth': emailAuthToken
+      }
+    })
+    nextModal()
+  }
+
+  const verifyCode = async () => {
+    const { verificationCode } = getValues()
+    await trigger('verificationCode')
+    if (!errors.verificationCode) {
+      try {
+        const response = await fetch(`${baseUrl}/email-auth/verify-pin`, {
+          method: 'POST',
+          headers: { 'Content-Type': 'application/json' },
+          credentials: 'include',
+          body: JSON.stringify({
+            pin: verificationCode,
+            email: formData.email
+          })
+        })
+        if (response.status === 201) {
+          setEmailVerified(true)
+          setCodeError('')
+          setEmailAuthToken(response.headers.get('email-auth') || '')
+        } else {
+          setCodeError('Verification code is not valid')
+          setEmailVerified(false)
+        }
+      } catch {
+        setEmailVerified(false)
+        setCodeError('Email verification failed')
+      }
```

Comment:
특별한 이유가 없다면 `fetch`보다 `safeFetcher`를 쓰는 게 나을 것 같아요! 
그리고 제가 신입부원 때 짠 코드의 영향이겠지만.... 지금 봤을 때 `codeError`라는 useState가 굳이 필요가 없어 보이네요..! `useForm<VerifyEmailInput>` 할 때 `setError`와 `clearErrors`도 포함해서 이런 식으로 처리하는 게 더 좋아 보여요! SignUpSendEmail.tsx의 `emailError`도 마찬가지고요!


```suggestion
      try {
        const response = await safeFetcher.post('email-auth/verify-pin', {
          json: {
            pin: verificationCode,
            email: formData.email
          },
          credentials: 'include'
        })
        setEmailVerified(true)
        clearErrors('verificationCode')
        setEmailAuthToken(response.headers.get('email-auth') || '')
      } catch (error) {
        setEmailVerified(false)
        if (isHttpError(error)) {
          setError('verificationCode', {
            message: 'Verification code is not valid'
          })
        } else {
          setError('verificationCode', {
            message: 'Email verification failed'
          })
        }
      }
```